registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -361,45 +467,112 @@ NormalDistributionsTransform<PointSource, PointTarget>::updateDerivatives (Eigen
 
 template<typename PointSource, typename PointTarget> void
 NormalDistributionsTransform<PointSource, PointTarget>::computeHessian (Eigen::Matrix<double, 6, 6> &hessian,
-                                                                        const PointCloudSource &trans_cloud)
+                                                                        const PointCloudSource &trans_cloud) const
 {
-  hessian.setZero ();
+  std::vector<Eigen::Matrix<double, 6, 6>, Eigen::aligned_allocator<Eigen::Matrix<double, 6, 6>>> hessians (num_threads_, Eigen::Matrix<double, 6, 6>::Zero());
 
   // Precompute Angular Derivatives unessisary because only used after regular derivative calculation
   // Update hessian for each point, line 17 in Algorithm 2 [Magnusson 2009]
+
+  // Parallel hessian calculation doesn't make the entire registration process fast
+  // because it is called much fewer times compared to derivative calculation (maybe)
+  // #pragma omp parallel for num_threads(num_threads_) schedule(guided, 8)
   for (std::size_t idx = 0; idx < input_->size (); idx++)
   {
+    Eigen::Matrix<double, 4, 6> point_jacobian = Eigen::Matrix<double, 4, 6>::Zero();
+    Eigen::Matrix<double, 24, 6> point_hessian = Eigen::Matrix<double, 24, 6>::Zero();
+    point_jacobian.block<3, 3>(0, 0).setIdentity();
+
+    Eigen::Matrix<double, 6, 6> pt_hessian = Eigen::Matrix<double, 6, 6>::Zero();
+  
     // Transformed Point
     const auto& x_trans_pt = trans_cloud[idx];
 
     // Find nieghbors (Radius search has been experimentally faster than direct neighbor checking.
     std::vector<TargetGridLeafConstPtr> neighborhood;
```

Comment:
Please move these also out of the for loop

Reply:
Done